app/src/main/java/com/sci/recipeandroid/feature/personalize/ui/adapter/DietRecipeAdapter.kt

```diff
@@ -0,0 +1,40 @@
+package com.sci.recipeandroid.feature.personalize.ui.adapter
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import com.sci.recipeandroid.databinding.ItemViewDietRecipeBinding
+import com.sci.recipeandroid.feature.personalize.domain.model.DietRecipeModel
+import com.sci.recipeandroid.feature.personalize.ui.viewholder.DietRecipeViewHolder
+
+class DietRecipeAdapter(private val onClickItem:(Int)->Unit): RecyclerView.Adapter<DietRecipeViewHolder>() {
+    private var List = emptyList<DietRecipeModel>()
+    private var selectedPosition = RecyclerView.NO_POSITION
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): DietRecipeViewHolder {
+        val binding = ItemViewDietRecipeBinding.inflate(LayoutInflater.from(parent.context),parent,false)
+        return DietRecipeViewHolder(binding,onClickItem)
+    }
+
+    override fun getItemCount(): Int {
+        return List.size
+    }
+
+    override fun onBindViewHolder(holder: DietRecipeViewHolder, position: Int) {
+        holder.bind(
+            List[position],position == selectedPosition
+        )
+        holder.itemView.setOnClickListener {
```

Comment:
please move view logic (ui bind , ui interaction) to view holder

Reply:
Updated in e2ceaf1519d2ad526bf9d419752063a820c4ad13